pkg/derivation/derivation.go

```diff
@@ -64,121 +88,43 @@ func (d *Derivation) Validate() error {
 		return fmt.Errorf("required attribute 'builder' missing")
 	}
 
-	for i, e := range d.EnvVars {
-		err := e.Validate()
-		if err != nil {
-			return fmt.Errorf("error validating env var '%s': %w", e.Key, err)
-		}
+	// there has to be an env variable with key `name`.
+	hasNameEnv := false
 
-		if i > 0 && e.Key < d.EnvVars[i-1].Key {
-			return fmt.Errorf("invalid env var order: %s < %s", e.Key, d.EnvVars[i-1].Key)
+	for k := range d.Env {
+		if k == "" {
+			return fmt.Errorf("empty environment variable key")
 		}
-	}
 
-	return nil
-}
-
-// WriteDerivation writes the textual representation of the derivation to the passed writer.
-func (d *Derivation) WriteDerivation(writer io.Writer) error {
-	outputs := make([][]byte, len(d.Outputs))
-	for i, o := range d.Outputs {
-		outputs[i] = encodeArray('(', ')', true, []byte(o.Name), []byte(o.Path), []byte(o.HashAlgorithm), []byte(o.Hash))
-	}
-
-	inputDerivations := make([][]byte, len(d.InputDerivations))
-	{
-		for i, in := range d.InputDerivations {
-			names := encodeArray('[', ']', true, stringsToBytes(in.Name)...)
-			inputDerivations[i] = encodeArray('(', ')', false, quoteString(in.Path), names)
+		if k == "name" {
+			hasNameEnv = true
 		}
 	}
 
-	envVars := make([][]byte, len(d.EnvVars))
-	{
-		for i, e := range d.EnvVars {
-			envVars[i] = encodeArray('(', ')', false, quoteString(e.Key), quoteString(e.Value))
-		}
+	if !hasNameEnv {
+		return fmt.Errorf("env 'name' not found")
 	}
 
-	_, err := writer.Write([]byte("Derive"))
-	if err != nil {
-		return err
-	}
-
-	_, err = writer.Write(
-		encodeArray('(', ')', false,
-			encodeArray('[', ']', false, outputs...),
-			encodeArray('[', ']', false, inputDerivations...),
-			encodeArray('[', ']', true, stringsToBytes(d.InputSources)...),
-			quoteString(d.Platform),
-			quoteString(d.Builder),
-			encodeArray('[', ']', true, stringsToBytes(d.Arguments)...),
-			encodeArray('[', ']', false, envVars...),
-		),
-	)
-
-	return err
+	return nil
 }
 
-// String returns the default (first) output path.
+// String returns the path of the derivation.
 func (d *Derivation) String() string {
-	return d.Outputs[0].Path
+	// TODO: replace with drv path
+	return "TODO"
```

Comment:
Maybe remove this for now? It's behaviour is currently incorrect anyway, and returning `TODO` is not useful.